pkg/util/steps/condition.go

```diff
@@ -5,13 +5,33 @@ package steps
 
 import (
 	"context"
+	"errors"
 	"fmt"
+	"net/http"
+	"strings"
 	"time"
 
 	"github.com/sirupsen/logrus"
 	"k8s.io/apimachinery/pkg/util/wait"
+
+	"github.com/Azure/ARO-RP/pkg/api"
 )
 
+// Functions that run as condition-steps return Error
+// instead of InternalServerError
+var timeoutConditionErrors = map[string]string{
```

Comment:
Looking for inputs on the error messages here, as these will customer visible when install/update/delete fails.
Also, I am cross-minded if we wanna share with customers saying `Hive deployment failed`, 

I want to make these errors not very generic, but more precise and actionable, the aim is to prevent InternalServerError with an actionable error or at least append `Please contact Support if retry fails`

Any thoughts?

Reply:
cc @bennerv @s-amann @cadenmarchese

Reply:
I have modified the code to prevent HiveRelated errors and only pass a generic timeout message, instead, please review.